tools/src/gen-params.rs

```diff
@@ -45,8 +41,32 @@ struct Args {
     params_root_dir: String,
 
     /// Generate Groth16 parameters from existing query parameters
-    #[arg(long)]
+    #[arg(long, default_value_t = false)]
     only_groth16: bool,
+
+    /// If this flag is true, then only the R1CS file is generated for Groth16
+    /// instead of all the Groth16 assets; useful when the proving and verification
+    /// keys must be generated with a trusted setup ceremony
+    #[arg(long, default_value_t = false)]
+    only_r1cs: bool,
+
+    #[command(subcommand)]
+    command: Option<Commands>,
```

Comment:
Could you please put both operation behind a `Command`? It's weird that PPs generation is nothing, but Solidity generation is gated behind a command.